### ISSUE TITLE
[BUG] SubscriptionClient stops processing message

### ISSUE THREAD

**Issue by aklefdal** [Azure/azure-sdk-for-net#15018]
**Describe the bug**
We have some clients receiving message from a topic that suddenly stops processing messages. The clients are Windows Services running on-premise, separate executables from several repositories. All of them are still running and doing other work like answering on an HTTP-endpoint. Not all clients fail at the same time, normally just one, but sometimes several clients at the same time.

![image](https://user-images.githubusercontent.com/404432/92600069-5e153900-f2ab-11ea-9aeb-3bdf66ca23b0.png)

**Expected behavior**
I expect the SubscriptionClient to continue processing messages as long as there are any available, and never stop. In case of network changes in Azure Service Bus I expect the client to gracefully handle that.

**Actual behavior (include Exception or Stack Trace)**
Message processing just stops, without exceptions being thrown.

**To Reproduce**
I cannot reproduce. It happens randomly.

**Environment:**
 - Microsoft.Azure.ServiceBus 4.1.4 
 - Windows Server 2012 R2, .NET Framework 4.7.2]

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team best able to assist.

**Comment by aklefdal** [comment]
Support ticket 120091324001127 has been created.

**Comment by aklefdal** [comment]
Microsoft.Azure.ServiceBus version 4.1.4 is unlisted. We are now running Microsoft.Azure.ServiceBus 4.1.3, and we are still having the same random problem.

**Comment by axisc** [comment]
@aklefdal, we will need to further investigate the backend for your namespace to see if this is caused due to a race condition.

Since this is not due to an issue with the SDK (or customer code) and is being tracked via the support ticket, I'm closing this here.

**Comment by aklefdal** [comment]
We have been looking more in to this problem and have found a working solution.

When message processing stopped, simultaneous in several jobs on the same machine, it was not because of trouble with the messaging part of the application.
* The problem was that the application hung on calls to Azure Blob Storage using the client (CloudBlobClient) in the SDK.
* We had not implemented any maximum execution time in this client.
* We did not, and still does not have, any maximum execution time for the message processing.
* When we added DefaultRequestOptions like this, we managed avoid stopping the message processing.
 
![image](https://user-images.githubusercontent.com/404432/93768239-6198b080-fc19-11ea-9268-887fd2ca9c23.png)

* We used to have RetryPolicy for our call to Upload, but not as default for all calls.
* We do not think the problem is with Azure Blob Storage or the CloudBlobClient, but rather with networking or VMWare on our side.

I am still curious to how I can have SubscriptionClient throw an exception in case of a timeout in the message processing. Other than that, this is no longer an issue for us.
